semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -77,6 +79,28 @@ trait TextDocumentOps {
       def addOccurrence(mpos: m.Position, gsym: g.Symbol, role: Role): Unit =
         addOccurrenceFromSemantic(mpos, gsym.toSemantic, role)
 
+      def addSamOccurrence(gtree: g.Function): Boolean = gtree.tpe match {
+        case tpe: g.TypeRef if tpe.args.isEmpty =>
+          val gpos = gtree.pos
+          val gsym = gtree.symbol
+          def atPos(mpos: m.Position): Boolean = gsym.toSemantic match {
+            case Symbols.None => false
+            case ssym =>
+              samoccurrences.update(mpos, (ssym, Role.DEFINITION))
+              if (!shouldNotSaveSymbol(gsym) && !shouldNotSaveSemanticSymbol(ssym))
+                saveSymbolFromSemantic(gsym.setInfo(tpe), ssym)
+              true
+          }
```

Comment:
```suggestion
      def addSamOccurrence(gtree: g.Function): Boolean = gtree.tpe match {
        case tpe: g.TypeRef if tpe.args.isEmpty =>
          val gpos = gtree.pos
          val gsym = gtree.symbol
          val gsymInfo = gtree.attachments.get[global.SAMFunction] match {
            case None => tpe
            case Some(global.SAMFunction(tpe, sym, sym2)) =>
              sym2.info
          }
          def atPos(mpos: m.Position): Boolean = gsym.toSemantic match {
            case Symbols.None => false
            case ssym =>
              samoccurrences.update(mpos, (ssym, Role.DEFINITION))
              if (!shouldNotSaveSymbol(gsym) && !shouldNotSaveSemanticSymbol(ssym))
                saveSymbolFromSemantic(gsym.setInfo(gsymInfo), ssym)
              true
          }
```
this will get the type to be the actual anonymous class that is generated

Reply:
We could also potentially use that  attachment instead of the other conditions to be sure we have a sam type

Reply:
done. thank you for finding this.